### PR TITLE
create models for VocabularyTopic and TaxonomyVocabulary

### PR DESCRIPTION
**Pull request by kaysiz**
- Setup new models for TaxonomyVocabulary and VocabularyTopic. 
- Setup admin interface. 
Next step
 - allow user to choose taxonomy entries in work form
 - enable user to see taxonomy details in work detail view
 - create API for listing taxonomies (/za/taxonomies/)

### CODE REVIEW

**Comment by kaysiz** [comment]
Need help writing tests for the Models, following the structure and convention already setup in the codebase.

**Comment by kaysiz** [comment]
@longhotsummer is this a good route to implement 'user can choose taxonomy entries in work form - EditWorkView', https://select2.org/getting-started/basic-usage. The multiple select option will work well with our use case.

File: indigo_api/models.py [hunk 1]
```diff
@@ -151,6 +151,28 @@ def get_queryset(self):
...
+    title = models.CharField(max_length=30, null=False, unique=True, blank=False)
+
+    def __unicode__(self):
+        return unicode(self.name)
+
+
+class VocabularyTopic(models.Model):
+    taxonomy_vocabulary = models.ForeignKey(TaxonomyVocabulary, related_name='vocabularies', null=False, blank=False, on_delete=models.CASCADE)
+    level_1 = models.CharField(max_length=30, null=False, unique=True, blank=False)
+    level_2 = models.CharField(max_length=30, null=False, unique=True, blank=False)
```
Comment (longhotsummer) [inline indigo_api/models.py]: level_2 can be null and black, and doesn't need to be unique.

File: indigo_api/models.py [hunk 2]
```diff
@@ -188,6 +210,9 @@ class Meta:
 
     stub = models.BooleanField(default=False, help_text="Stub works do not have content or points in time")
 
+    # taxonomies
+    taxonomies = models.ManyToManyField(VocabularyTopic, related_name='vocabulary_topics')
```
Comment (longhotsummer) [inline indigo_api/models.py]:
```suggestion
    taxonomies = models.ManyToManyField(VocabularyTopic, related_name='works')
```

**Comment by longhotsummer** [comment]
@kaysiz looking good, some minor comments.

For the UI, we currently use https://developer.snapappointments.com/bootstrap-select/examples/ which is similar to select2, which I think will be good enough.

File: indigo_api/models.py (hunk 1 above)
Comment (kaysiz) [reply indigo_api/models.py]: level_1 cannot be null and blank?

Comment (longhotsummer) [reply indigo_api/models.py]: Correct: level_1 is not optional. Level 2 IS optional. Neither need to be unique on their own, only when combined together using the `unique_together` index below.

**Comment by longhotsummer** [comment]
Replaced by #625.

**Pull request closed without merging**